Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -1418,26 +1383,187 @@ class Gx2Fitter {
         break;
       }
 
-      if (multipleScattering) {
-        // update the scattering angles
-        for (std::size_t matSurface = 0; matSurface < nMaterialSurfaces;
-             matSurface++) {
-          const std::size_t deltaPosition = eBoundSize + 2 * matSurface;
-          const GeometryIdentifier geoId = geoIdVector[matSurface];
-          const auto scatteringMapId = scatteringMap.find(geoId);
-          assert(scatteringMapId != scatteringMap.end() &&
-                 "No scattering angles found for material surface.");
-          scatteringMapId->second.scatteringAngles().block<2, 1>(2, 0) +=
-              deltaParamsExtended.block<2, 1>(deltaPosition, 0).eval();
-        }
-      }
-
       oldChi2sum = extendedSystem.chi2();
     }
     ACTS_DEBUG("Finished to iterate");
     ACTS_VERBOSE("Final parameters: " << params.parameters().transpose());
     /// Finish Fitting /////////////////////////////////////////////////////////
 
+    /// Actual MATERIAL Fitting ////////////////////////////////////////////////
+    ACTS_DEBUG("Start to evaluate material");
+    if (multipleScattering) {
```

Comment:
I think this function is quite long by now and should be split apart. I would keep propagator stuff here but push some computations out. For example the whole material fitting here is >100 lines and could be extracted to its own function.

Reply:
cleaned out in:
- https://github.com/acts-project/acts/pull/3839
- https://github.com/acts-project/acts/pull/3840